test/Shared/UserManagerTestBase.cs

```diff
@@ -83,15 +83,10 @@ protected RoleManager<TRole> CreateRoleManager(object context = null, IServiceCo
         protected abstract void AddUserStore(IServiceCollection services, object context = null);
         protected abstract void AddRoleStore(IServiceCollection services, object context = null);
 
-        protected TUser CreateTestUser(string namePrefix = "")
-        {
-            return new TUser() { UserName = namePrefix + Guid.NewGuid().ToString() };
-        }
+        protected abstract TUser CreateTestUser(string username = "", string email = "", string phoneNumber = "",
+            bool lockoutEnabled = false, DateTimeOffset? lockoutEnd = null);
```

Comment:
I kind of liked this that took the parameter as username and appended Guid when calling this. I feel strongly about keeping it this way since it more cleaner. 

Do you feel we are creating a pit of failure here for users ? They can always refer to test code for usage ?